File: ORB.Core/Models/UserPartnerOrganisation.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class UserPartnerOrganisation : EntityWithEditor
    {
        public int UserID { get; set; }
        public int PartnerOrganisationID { get; set; }
        public bool IsAdmin { get; set; }
        public bool? HideHeadlines { get; set; }
        public bool? HideMilestones { get; set; }
        public bool? HideCustomSections { get; set; }

        public virtual PartnerOrganisation PartnerOrganisation { get; set; }
        public virtual User User { get; set; }
    }
}

File: corporate-reporting/src/types/UserPartnerOrganisation.ts
import { IEntity, Entity } from "./Entity";
import { IUser } from "./User";
import { IPartnerOrganisation } from "./PartnerOrganisation";

export interface IUserPartnerOrganisation extends IEntity {
    UserID: number;
    PartnerOrganisationID: number;
    IsAdmin: boolean;
    HideHeadlines: boolean;
    HideMilestones: boolean;
    HideCustomSections: boolean;
    User?: IUser;
    PartnerOrganisation?: IPartnerOrganisation;
}

export class UserPartnerOrganisation extends Entity implements IUserPartnerOrganisation {
    public UserID = null;
    public PartnerOrganisationID = null;
    public IsAdmin = false;
    public HideHeadlines = false;
    public HideMilestones = false;
    public HideCustomSections = false;
}